packages/invoices/src/OutboundShipment/DetailView/Toolbar.tsx

```diff
@@ -5,19 +5,52 @@ import {
   InputWithLabelRow,
   BasicTextInput,
   Grid,
+  DropdownMenu,
+  DropdownMenuItem,
+  DeleteIcon,
+  useTranslation,
+  useNotification,
+  useTableStore,
 } from '@openmsupply-client/common';
 import { CustomerSearchInput } from '../CustomerSearchInput';
-import { OutboundShipment } from './types';
+import { ItemRow, OutboundShipment } from './types';
 import { isInvoiceEditable } from '../utils';
 
 interface ToolbarProps {
   draft: OutboundShipment;
 }
 
 export const Toolbar: FC<ToolbarProps> = ({ draft }) => {
+  const t = useTranslation();
+  const { success, info } = useNotification();
+
+  const { selectedRows } = useTableStore(state => ({
+    selectedRows: Object.keys(state.rowState)
+      .filter(id => state.rowState[id]?.isSelected)
+      .map(selectedId => draft.lines.find(({ id }) => selectedId === id))
+      .filter(Boolean) as ItemRow[],
+  }));
+
+  const deleteAction = () => {
+    if (selectedRows && selectedRows?.length > 0) {
+      selectedRows.forEach(line => draft.deleteLine?.(line));
+      const successSnack = success(`Deleted ${selectedRows?.length} lines`);
+      successSnack();
+    } else {
+      const infoSnack = info('Select rows to delete them');
+      infoSnack();
```

Comment:
great names - liking the `infoSnack` and `successSnack`! 😄 
:nit just noticed after commenting.. translation for the message?